File: app/src/main/java/com/kunzisoft/keepass/activities/ReadOnlyHelper.java
package com.kunzisoft.keepass.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import com.kunzisoft.keepass.settings.PreferencesUtil;

public class ReadOnlyHelper {

    public static final String READ_ONLY_KEY = "READ_ONLY_KEY";

    public static final boolean READ_ONLY_DEFAULT = false;

    public static boolean retrieveReadOnlyFromInstanceStateOrPreference(Context context, Bundle savedInstanceState) {
        boolean readOnly;
        if (savedInstanceState != null
                && savedInstanceState.containsKey(READ_ONLY_KEY)) {
            readOnly = savedInstanceState.getBoolean(READ_ONLY_KEY);
        } else {
            readOnly = PreferencesUtil.enableReadOnlyDatabase(context);
        }
        return readOnly;
    }

    public static boolean retrieveReadOnlyFromInstanceStateOrArguments(Bundle savedInstanceState, Bundle arguments) {
        boolean readOnly = READ_ONLY_DEFAULT;
        if (savedInstanceState != null
                && savedInstanceState.containsKey(READ_ONLY_KEY)) {
            readOnly = savedInstanceState.getBoolean(READ_ONLY_KEY);
        } else if (arguments != null
                    && arguments.containsKey(READ_ONLY_KEY)) {
                readOnly = arguments.getBoolean(READ_ONLY_KEY);
        }
        return readOnly;
    }

    public static boolean retrieveReadOnlyFromInstanceStateOrIntent(Bundle savedInstanceState, Intent intent) {
        boolean readOnly = READ_ONLY_DEFAULT;
        if (savedInstanceState != null
                && savedInstanceState.containsKey(READ_ONLY_KEY)) {
            readOnly = savedInstanceState.getBoolean(READ_ONLY_KEY);
        } else {
            if (intent != null)
                readOnly = intent.getBooleanExtra(READ_ONLY_KEY, READ_ONLY_DEFAULT);
        }
        return readOnly;
    }

    public static void putReadOnlyInIntent(Intent intent, boolean readOnly) {
        intent.putExtra(READ_ONLY_KEY, readOnly);
    }

    public static void putReadOnlyInBundle(Bundle bundle, boolean readOnly) {
        bundle.putBoolean(READ_ONLY_KEY, readOnly);
    }

    public static void onSaveInstanceState(Bundle outState, boolean readOnly) {
        outState.putBoolean(READ_ONLY_KEY, readOnly);
    }
}

File: app/src/main/java/com/kunzisoft/keepass/activities/helpers/ReadOnlyHelper.kt
/*
 * Copyright 2019 Jeremy Jamet / Kunzisoft.
 *
 * This file is part of KeePassDX.
 *
 *  KeePassDX is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  KeePassDX is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with KeePassDX.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.kunzisoft.keepass.activities.helpers

import android.content.Context
import android.content.Intent
import android.os.Bundle

import com.kunzisoft.keepass.settings.PreferencesUtil

object ReadOnlyHelper {

    private const val READ_ONLY_KEY = "READ_ONLY_KEY"

    const val READ_ONLY_DEFAULT = false

    fun retrieveReadOnlyFromIntent(intent: Intent): Boolean {
        return intent.getBooleanExtra(READ_ONLY_KEY, READ_ONLY_DEFAULT)
    }

    fun retrieveReadOnlyFromInstanceStateOrPreference(context: Context, savedInstanceState: Bundle?): Boolean {
        return if (savedInstanceState != null && savedInstanceState.containsKey(READ_ONLY_KEY)) {
            savedInstanceState.getBoolean(READ_ONLY_KEY)
        } else {
            PreferencesUtil.enableReadOnlyDatabase(context)
        }
    }

    fun retrieveReadOnlyFromInstanceStateOrArguments(savedInstanceState: Bundle?, arguments: Bundle?): Boolean {
        var readOnly = READ_ONLY_DEFAULT
        if (savedInstanceState != null && savedInstanceState.containsKey(READ_ONLY_KEY)) {
            readOnly = savedInstanceState.getBoolean(READ_ONLY_KEY)
        } else if (arguments != null && arguments.containsKey(READ_ONLY_KEY)) {
            readOnly = arguments.getBoolean(READ_ONLY_KEY)
        }
        return readOnly
    }

    fun retrieveReadOnlyFromInstanceStateOrIntent(savedInstanceState: Bundle?, intent: Intent?): Boolean {
        var readOnly = READ_ONLY_DEFAULT
        if (savedInstanceState != null && savedInstanceState.containsKey(READ_ONLY_KEY)) {
            readOnly = savedInstanceState.getBoolean(READ_ONLY_KEY)
        } else {
            if (intent != null)
                readOnly = intent.getBooleanExtra(READ_ONLY_KEY, READ_ONLY_DEFAULT)
        }
        return readOnly
    }

    fun putReadOnlyInIntent(intent: Intent, readOnly: Boolean) {
        intent.putExtra(READ_ONLY_KEY, readOnly)
    }

    fun putReadOnlyInBundle(bundle: Bundle, readOnly: Boolean) {
        bundle.putBoolean(READ_ONLY_KEY, readOnly)
    }

    fun onSaveInstanceState(outState: Bundle, readOnly: Boolean) {
        outState.putBoolean(READ_ONLY_KEY, readOnly)
    }
}
